src/theory/strings/theory_strings_rewriter.h

```diff
@@ -28,7 +28,34 @@ namespace theory {
 namespace strings {
 
 class TheoryStringsRewriter {
-private:
+ private:
+  /** simple regular expression consume
+   *
+   * This method is called when we are rewriting a membership of the form
+   *   s1 ++ ... ++ sn in r1 ++ ... ++ rm
+   * We have that mchildren consists of the strings s1...sn, and children
+   * consists of the regular expressions r1...rm.
+   *
+   * This method tries to strip off parts of the concatenation terms. It updates
+   * the vectors such that the resulting vectors are such that the membersip
+   * mchildren[n'...n''] in children[m'...m''] is equivalent to the input
+   * membership. The argument dir indicates the direction to consider, where
+   * 0 means strip off the front, 1 off the back, and < 0 off of both.
+   *
+   * If this method returns the false node, then we have inferred that the input
+   * membership is equivalent to false.
+   *
+   * For example, given input
+   *   mchildren = { "ab", x }, children = { "a", ("cd")* } and dir = 0,
```

Comment:
Shouldn't it be [["a"]] and [["cd"]] on the right-hand side?

Reply:
Yes, I've fixed.